docs/programming/profiling-nvidia-gpu-performance.md

```diff
@@ -0,0 +1,1157 @@
+      0.0              441          6          73.5         50.0        50          181          52.8  fflush
+
+Processing [report.sqlite] with [/curc/sw/cuda/12.1.1/nsight-systems-2023.1.2/host-linux-x64/reports/cuda_api_sum.py]...
+
+ ** CUDA API Summary (cuda_api_sum):
+
+ Time (%)  Total Time (ns)  Num Calls    Avg (ns)     Med (ns)    Min (ns)    Max (ns)    StdDev (ns)            Name
+ --------  ---------------  ---------  ------------  -----------  ---------  -----------  ------------  ----------------------
+     96.4      139,706,807          3  46,568,935.7      3,817.0      3,035  139,699,955  80,653,828.6  cudaMalloc
+      3.5        5,038,894          1   5,038,894.0  5,038,894.0  5,038,894    5,038,894           0.0  cudaLaunchKernel
+      0.1          160,483          3      53,494.3      7,274.0      2,966      150,243      83,814.5  cudaFree
+      0.0           65,401          3      21,800.3     25,317.0      8,315       31,769      12,116.0  cudaMemcpy
+      0.0            1,333          1       1,333.0      1,333.0      1,333        1,333           0.0  cuModuleGetLoadingMode
+
+Processing [report.sqlite] with [/curc/sw/cuda/12.1.1/nsight-systems-2023.1.2/host-linux-x64/reports/cuda_gpu_kern_sum.py]...
+
+ ** CUDA GPU Kernel Summary (cuda_gpu_kern_sum):
+
+ Time (%)  Total Time (ns)  Instances  Avg (ns)  Med (ns)  Min (ns)  Max (ns)  StdDev (ns)                  Name
+ --------  ---------------  ---------  --------  --------  --------  --------  -----------  ------------------------------------
+    100.0            2,112          1   2,112.0   2,112.0     2,112     2,112          0.0  vectorAdd(float *, float *, float *)
+
+Processing [report.sqlite] with [/curc/sw/cuda/12.1.1/nsight-systems-2023.1.2/host-linux-x64/reports/cuda_gpu_mem_time_sum.py]...
+
+ ** CUDA GPU MemOps Summary (by Time) (cuda_gpu_mem_time_sum):
+
+ Time (%)  Total Time (ns)  Count  Avg (ns)  Med (ns)  Min (ns)  Max (ns)  StdDev (ns)      Operation
+ --------  ---------------  -----  --------  --------  --------  --------  -----------  ------------------
+     52.2            2,656      2   1,328.0   1,328.0     1,216     1,440        158.4  [CUDA memcpy HtoD]
+     47.8            2,432      1   2,432.0   2,432.0     2,432     2,432          0.0  [CUDA memcpy DtoH]
+
+Processing [report.sqlite] with [/curc/sw/cuda/12.1.1/nsight-systems-2023.1.2/host-linux-x64/reports/cuda_gpu_mem_size_sum.py]...
+
+ ** CUDA GPU MemOps Summary (by Size) (cuda_gpu_mem_size_sum):
+
+ Total (MB)  Count  Avg (MB)  Med (MB)  Min (MB)  Max (MB)  StdDev (MB)      Operation
+ ----------  -----  --------  --------  --------  --------  -----------  ------------------
+      0.008      2     0.004     0.004     0.004     0.004        0.000  [CUDA memcpy HtoD]
+      0.004      1     0.004     0.004     0.004     0.004        0.000  [CUDA memcpy DtoH]
+ 
+```
+::::
+
+### Interpreting Key Outputs
```

Comment:
```suggestion
The provided nsys report contains key output that can help you optimize your code. For more information on these key output sections and interpreting the output, please review the tabs below.
```
For the other tools, we do not create a subsection for interpreting the output. So, I think we should remove it and replace it with my suggestion.